### ISSUE TITLE
API key not provided anymore by casambi support

### ISSUE THREAD

**Issue by BastianZim** [awahlig/homebridge-casambi#32]
This is not really a support request, more an update as maybe someone has an idea for a workaround.

I contacted casambi support yesterday to request an API key and was informed that they do not provide them anymore, as the project is in beta (not sure why that is a reason).

At some point they might again, but they do not have an ETA and would not commit to anything.

**Comment by awahlig** [comment]
That explanation is surprising as the API has always been in beta and the documentation still states that a key can be obtained by contacting the support.

My key expires soon so I just requested an extension, will report back what they said.

**Comment by BastianZim** [comment]
Yeah, I don't understand it either, especially since it's not mentioned anywhere and, as you wrote, the website says to contact them.

Anyways this was their exact response:

> Unfortunately, due to limited resources and being API-Key in Beta-program, we are not able to issue new API Keys.
This can change in future. I kindly ask you to follow the situation either by contacting us again within some time or keep an eye on our website updates.

> My key expires soon so I just requested an extension, will report back what they said.

Maybe you have more luck since you're just renewing?

**Comment by awahlig** [comment]
Yeah, I got an extension no problem.

I asked a follow-up question about new users, waiting for a reply.

**Comment by awahlig** [comment]
Sadly it looks like they indeed closed it for new users for now.

Here's the answer I got:
> Unfortunately we cannot issue more API keys for domestic use due to limited server resources under the BETA program. The situation may change in the near future, which is why we recommend following the situation. At the moment we don't have an exact date of when the API keys will become available again.

Basically, the same answer you got.

**Comment by m-a-r-c-u-s** [comment]
Even though Casambi wont generate new API Keys some hope for the future. Casambi will support Matter probably via the newly launched cloud gateway. This is just a Rasberry Pi 4 running on custom OS which means they might make the ISO available for download if you already own a RPi.

https://casambi.com/news/casambi-joins-csa-alliance-confirming-support-for-matter/

https://casambi.com/ecosystem/casambi-cloud-gateway/

**Comment by BastianZim** [comment]
So strange, because their reply was again that I should monitor the website but the website still has the old statement.

> Casambi will support Matter probably via the newly launched cloud gateway.

Have asked that and about the Raspberry Pi, let's see.

I'm wondering though if anyone has tried to connect using EnOcean? It's supported by casambi and HomeBridge: https://casambi.com/news/marketing-enoceanperfectfit/

**Comment by BastianZim** [comment]
Have heard back. 

Matter is in development but they don't have an ETA yet.
API keys will (probably) be distributed again but not now. I guess, once they're out of beta.

The availability of API keys will be announced on their website and you can also contact support.

Seems like there is currently no way to add casambi to Apple HomeKit, if you don't already have an API key.

**Comment by BastianZim** [comment]
Another update in case someone finds this thread: API keys for personal use are still not available, but API keys for companies are. Just apply as a company, and you'll get the key which will be valid for 100 years (yes).

**Comment by BoldBowl** [comment]
> Another update in case someone finds this thread: API keys for personal use are still not available, but API keys for companies are. Just apply as a company, and you'll get the key which will be valid for 100 years (yes).

Thanks for the suggestion, how did you request an API-key “as a company”? Thanks!!

**Comment by BastianZim** [comment]
I mean applying as a company, not an individual person. But that requires that you have a company.
